src/imperative/cached_op.cc

```diff
@@ -290,7 +290,7 @@ bool CachedOp::CheckDynamicShapeExists(const Context& default_ctx,
   CheckAndInferShape(&g, std::move(shape_inputs), true,
                      {0, 0}, {0, 0},
                      &contain_dynamic_shape);
-  if (contain_dynamic_shape && erase_result) {
+  if (!config_.static_shape && erase_result) {
```

Comment:
Sounds like something wasn't properly tested here? Maybe add a test to cover this path?

Reply:
It's already covered by current UT, there're 2 cases failed before fixing this.